tree/ntuple/v7/test/ntuple_types.cxx

```diff
@@ -274,6 +425,39 @@ TEST(RNTuple, Double)
    EXPECT_DOUBLE_EQ(2.0, *reader->GetModel()->GetDefaultEntry()->Get<double>("d2"));
 }
 
+TEST(RNTuple, Float)
+{
+   FileRaii fileGuard("test_ntuple_float.root");
+
+   auto model = RNTupleModel::Create();
+
+   auto f1 = std::make_unique<RField<float>>("f1");
+   f1->SetColumnRepresentative({ROOT::Experimental::EColumnType::kReal32});
+   model->AddField(std::move(f1));
+
+   auto f2 = std::make_unique<RField<float>>("f2");
+   f2->SetColumnRepresentative({ROOT::Experimental::EColumnType::kSplitReal32});
+   model->AddField(std::move(f2));
+
+   {
+      auto writer = RNTupleWriter::Recreate(std::move(model), "ntuple", fileGuard.GetPath());
+      auto e = writer->CreateEntry();
+      *e->Get<float>("f1") = 1.0;
+      *e->Get<float>("f2") = 2.0;
+      writer->Fill(*e);
+   }
+
+   auto reader = RNTupleReader::Open("ntuple", fileGuard.GetPath());
+   const auto *desc = reader->GetDescriptor();
+   EXPECT_EQ(ROOT::Experimental::EColumnType::kReal32,
+             (*desc->GetColumnIterable(desc->FindFieldId("f1")).begin()).GetModel().GetType());
+   EXPECT_EQ(ROOT::Experimental::EColumnType::kSplitReal32,
+             (*desc->GetColumnIterable(desc->FindFieldId("f2")).begin()).GetModel().GetType());
+   reader->LoadEntry(0);
+   EXPECT_FLOAT_EQ(1.0, *reader->GetModel()->GetDefaultEntry()->Get<float>("f1"));
+   EXPECT_FLOAT_EQ(2.0, *reader->GetModel()->GetDefaultEntry()->Get<float>("f2"));
```

Comment:
This test (and other) seems to check that the value can be written and read back as expected.  Is there a part of the test that check that the data is going through the intended encoding (as opposed to the wrong-but-still-reversible encoding)? I.e. maybe check the size of the output?

Reply:
This is partially tested in `ntuple_endian.cxx` (see e.g. [here](https://github.com/root-project/root/pull/12220/files#diff-f62365121555374687a4f7b26ce371458aebad84866d1a84690ad8e1ac2e1a80R191-R203)).